RecoParticleFlow/PFClusterTools/src/PFPhotonClusters.cc

```diff
@@ -157,11 +157,9 @@ void PFPhotonClusters::FillClusterShape() {
       if (std::abs(deta) > 2 || std::abs(dphi) > 2)
         continue;
 
-      //if(std::abs(dphi)<=2 && std::abs(deta)<=2){
       EBDetId EBidSeed = EBDetId(idseed_.rawId());
       EBDetId EBid = EBDetId(id.rawId());
-      int ind1 = EBidSeed.ieta() - EBid.ieta();
-      int ind2 = EBidSeed.iphi() - EBid.iphi();
+      int ind1;
       if (EBidSeed.ieta() * EBid.ieta() > 0) {
         ind1 = EBid.ieta() - EBidSeed.ieta();
```

Comment:
Sometime these warnings raised by the static analyzer are able to pinpoints small mistakes in the code.
In this case, I have the impression that here it should be
```C++
        ind1 = EBidSeed.ieta() - EBid.ieta();
```
and not its opposite.

Of course, if the exact distribution of energy inside the ```e5x5_``` matrix is not needed, this doesn't matter. But if that shape is needed anywhere, this could be a bug.

@Sam-Harper , could you please have a glance and tell me?

Reply:
likely yes. However this is dead code. Can we delete this class?  How best, you make this PR and then I submit the deletion on top of it?

Best,
Sam

Reply:
Thank you @Sam-Harper : cleaning is always welcome!
I can even do it since now here, if you agree, and then after this PR is merged you can check whether there are other obsolete classes that can also get cleaned. I see that the header of this class has to be removed also from RecoParticleFlow/PFProducer/src/PFEGammaAlgo.cc (where it does nothing, and therefore it can be safely removed)

Reply:
>I can even do it since now here, if you agree,

even better, more than happy for you to go ahead and remove. 

> and then after this PR is merged you can check whether there are other obsolete classes that can also get cleaned. 

There almost certainly are. And while I really really want to do this, I'm currently forbidden from contributing to e/gamma code  until I finish writing my sections of the e/gamma paper. Have to eat my vegetables before I can have desert.